components/base/sensorcontrolled/velocities.go

```diff
@@ -56,8 +52,48 @@ func (sb *sensorBase) SetVelocity(
 	var sensorCtx context.Context
 	sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
 
-	// TODO: RSDK-5355 remove control loop bool after testing
-	if useControlLoop && sb.loop != nil {
+	if useControlLoop {
+		// stop and restart loop
+		if sb.loop != nil {
+			if err := sb.Stop(ctx, nil); err != nil {
+				sb.logger.Error(err)
+			}
+		}
+		loop, err := control.NewLoop(sb.logger, controlLoopConfig, sb)
```

Comment:
When `base.Stop` was called the wheels weren't stopping because the loop was still running, so I added `loop.Stop` to the `base.Stop` code. so the loop needs to be started again when `SetVelocity` is called. Here I wanted to check if there was already a loop and if so stop it so we wouldn't have multiple loops running at the same time.

I can try instead of stopping the loop just changing the setpoint block to 0 when `base.Stop` is called. I tried doing this with encoded motor and ran into issues but I have a better understanding of all this stuff now so I can try again

Reply:
okay so just tried the setting to 0 thing. definitely works but still the same issue I had with motor where then the motor is just on but not moving, and the only way to turn the motor off is to stop the server

Reply:
ok make sense. Something we should track and fix in the control loop at a later stage

Reply:
there's a ticket to use the contexts in the control loop to stop them in our backlog. ANother thing would be to toy around with setting the velocities to zero through an update loop command, but I'd rather leave that as an incremental next step.